spec/client_spec.rb

```diff
@@ -367,6 +367,111 @@ def redis.with
     end
   end
 
+  describe 'get_remaining_ttl_for_resource' do
+    context 'when lock is valid' do
+      after(:each) { lock_manager.unlock(@lock_info) if @lock_info }
+
+      it 'gets the remaining ttl of a lock' do
+        ttl = 20_000
+        @lock_info = lock_manager.lock(resource_key, ttl)
+        remaining_ttl = lock_manager.get_remaining_ttl_for_resource(resource_key)
+        expect(remaining_ttl).to be_within(300).of(ttl)
+      end
+
+      context 'when servers respond with varying ttls' do
+        let (:servers) { ["redis://#{redis1_host}:#{redis1_port}", "redis://#{redis2_host}:#{redis2_port}"] }
+        let (:redlock) { Redlock::Client.new(servers) }
+        after(:each) { redlock.unlock(@lock_info) if @lock_info }
+
+        it 'returns the minimum ttl value' do
```

Comment:
@maltoe there's also a little bit of gymnastics going on here to test this, but it should be correct. Let me know what you think.

Reply:
🤔  yes, would be nicer if the "calculation" part of it (so everything besides the actual redis call) was a pure public function on some module... but again, I think if you start refactoring things like this, it'll be very difficult to stop at some point 😄 

would it be difficult to add a third server that responds with `15_000` and then see that this is returned (for the `.last(@quorum).min` line)?